### PR TITLE
Add support for clusterawsadm

### LINKED ISSUE #803: Add Support for clusterawsadm
## Expected Behaviour

Add Support for clusterawsadm CLI

## Current Behaviour
NA

## Are you a GitHub Sponsor yet (Yes/No?)

- [ ] Yes
- [X] No

## If requesting a CLI for "arkade get"

How many downloads does this tool have? Check at: 

https://somsubhra.github.io/github-release-stats/

  Total Downloads 62,680

### PR DESCRIPTION
**Pull request by tuxtof**
Signed-off-by: Christophe Jauffret <reg-github@geo6.net>

## Description

Add support for `clusterawsadm` CLI in arkade

## Motivation and Context

- [X] I have raised an issue to propose this change, which has been given a label of `design/approved` by a maintainer ([required](https://github.com/alexellis/arkade/blob/master/CONTRIBUTING.md))
Closes #803 

## How Has This Been Tested?

`./hack/test-tool.sh clusterawsadm`

```
+ go build
+ ./arkade get clusterawsadm --arch arm64 --os darwin --quiet
+ file /Users/cris/.arkade/bin/clusterawsadm
/Users/cris/.arkade/bin/clusterawsadm: Mach-O 64-bit executable arm64
+ rm /Users/cris/.arkade/bin/clusterawsadm
+ ./arkade get clusterawsadm --arch x86_64 --os darwin --quiet
+ file /Users/cris/.arkade/bin/clusterawsadm
/Users/cris/.arkade/bin/clusterawsadm: Mach-O 64-bit executable x86_64
+ rm /Users/cris/.arkade/bin/clusterawsadm
+ ./arkade get clusterawsadm --arch x86_64 --os linux --quiet
+ file /Users/cris/.arkade/bin/clusterawsadm
/Users/cris/.arkade/bin/clusterawsadm: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, Go BuildID=5I19QLJndROz4NKhj7_m/kBUP9uPqIFkhZp9k8H6Y/c5SlHMf_qRH6yFdRwcoU/UiBllQtCPcLE1ouQlwxh, not stripped
+ rm /Users/cris/.arkade/bin/clusterawsadm
+ ./arkade get clusterawsadm --arch arm64 --os linux --quiet
+ file /Users/cris/.arkade/bin/clusterawsadm
/Users/cris/.arkade/bin/clusterawsadm: ELF 64-bit LSB executable, ARM aarch64, version 1 (SYSV), statically linked, Go BuildID=mRb7O0ywcuu3ZHT3W__T/pwlBk88vCaSdJMB9AZ_D/1K7gEaq233w_VSYbGODx/CjpssFBE8kPGeCpKZv6P, not stripped
+ rm /Users/cris/.arkade/bin/clusterawsadm
```

`make test`

```
CGO_ENABLED=0 go test github.com/alexellis/arkade github.com/alexellis/arkade/cmd github.com/alexellis/arkade/cmd/apps github.com/alexellis/arkade/cmd/chart github.com/alexellis/arkade/cmd/system github.com/alexellis/arkade/pkg github.com/alexellis/arkade/pkg/apps github.com/alexellis/arkade/pkg/archive github.com/alexellis/arkade/pkg/config github.com/alexellis/arkade/pkg/env github.com/alexellis/arkade/pkg/get github.com/alexellis/arkade/pkg/helm github.com/alexellis/arkade/pkg/k8s github.com/alexellis/arkade/pkg/types -cover
?   	github.com/alexellis/arkade	[no test files]
ok  	github.com/alexellis/arkade/cmd	(cached)	coverage: 17.2% of statements
ok  	github.com/alexellis/arkade/cmd/apps	(cached)	coverage: 1.9% of statements
?   	github.com/alexellis/arkade/cmd/chart	[no test files]
?   	github.com/alexellis/arkade/cmd/system	[no test files]
?   	github.com/alexellis/arkade/pkg	[no test files]
?   	github.com/alexellis/arkade/pkg/apps	[no test files]
?   	github.com/alexellis/arkade/pkg/archive	[no test files]
ok  	github.com/alexellis/arkade/pkg/config	(cached)	coverage: 22.9% of statements
?   	github.com/alexellis/arkade/pkg/env	[no test files]
ok  	github.com/alexellis/arkade/pkg/get	(cached)	coverage: 52.8% of statements
ok  	github.com/alexellis/arkade/pkg/helm	(cached)	coverage: 5.3% of statements
?   	github.com/alexellis/arkade/pkg/k8s	[no test files]
?   	github.com/alexellis/arkade/pkg/types	[no test files]
```

`make e2e`

```
CGO_ENABLED=0 go test github.com/alexellis/arkade/pkg/get -cover --tags e2e -v
......
PASS
coverage: 55.1% of statements
ok  	github.com/alexellis/arkade/pkg/get	(cached)	coverage: 55.1% of statements

```

## Are you a GitHub Sponsor yet (Yes/No?)

- [ ] Yes
- [X] No

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Documentation

- [X] I have updated the list of tools in README.md if (required) with `./arkade get -o markdown`
- [ ] I have updated the list of apps in README.md if (required) with `./arkade install --help`

## Checklist:
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I've read the [CONTRIBUTION](https://github.com/alexellis/arkade/blob/master/CONTRIBUTING.md) guide
- [X] I have signed-off my commits with `git commit -s`

- [ ] I have tested this on arm, or have added code to prevent deployment

### CODE REVIEW

**Comment by Jasstkn** [comment]
@tuxtof hi. would you mind to update your branch from master? I'll review your PR then.

**Review by Jasstkn — COMMENTED** [review]
Hi. Thank you for your contribution!

I have one question: why you didn't include windows x amd64 in the config? As far as I can see, it's being uploaded on the Releases page:
e.g.
https://github.com/kubernetes-sigs/cluster-api-provider-aws/releases/download/v2.0.2/clusterawsadm-windows-amd64.exe

**Comment by tuxtof** [comment]
Hi @Jasstkn 
At the time i write the PR in v1.5, windows version was not available
Now with v2.0 it is so i just fix

File: pkg/get/get_test.go [hunk 1]
```diff
@@ -5537,3 +5537,61 @@ func Test_DownloadGrype(t *testing.T) {
...
+			url:     `https://github.com/kubernetes-sigs/cluster-api-provider-aws/releases/download/v2.0.2/clusterawsadm-darwin-arm64`,
+		},
+		{
+			os:      "ming",
+			arch:    arch64bit,
+			version: toolVersion,
+			url:     `https://github.com/kubernetes-sigs/cluster-api-provider-aws/releases/download/v2.0.2/clusterawsadm-windows-amd64.exe`,
+		},
+		{
+			os:      "ming",
```
Comment (Jasstkn) [inline pkg/get/get_test.go:5581]: only ming x amd64 is supported by arkade. you can remove ming x arm combination

**Review by Jasstkn — CHANGES_REQUESTED** [review]

File: pkg/get/get_test.go (hunk 1 above)
Comment (alexellis) [reply pkg/get/get_test.go:5581]: Please ping me by name when this is done, and I'll merge.

**Comment by Jasstkn** [comment]
@tuxtof I've fixed error in e2e pipeline, you can update your branch from master to make it go away.

**Comment by alexellis** [comment]
Let's merge it as is - then we can remove the Windows ARM64 binary ourselves and get this PR closed out.

**Pull request merged**